### ISSUE TITLE
Set namespace for TLS secretName in IngressRoute

### ISSUE THREAD

**Issue by MikaelElkiaer** [containous/traefik#5736]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

For IngressRoute - I would like to be able to set namespace for TLS secretName.
Currently, I am putting my IngressRoutes into the same namespace as the service they refer to.
And it is currently not possible to point the TLS secretName at another namespace.
So if I have more than one IngressRoute using the same certificate, then I need to create a secret for every namespace.

**Comment by dtomcej** [comment]
Secrets are namespaced for isolation and security reasons. We should not support cross-namespace secret mounting, as it would allow users to fish for secrets.

**Comment by MikaelElkiaer** [comment]
> Secrets are namespaced for isolation and security reasons. We should not support cross-namespace secret mounting, as it would allow users to fish for secrets.

Fair point.

Can anyone say how I should go about maintaining a certificate that I'd like to use for IngressRoutes in multiple namespaces?

**Comment by tlvenn** [comment]
@dtomcej I actually dont think it's a fair point ;)

With wildcard certificates, this is becoming quite a headache and right now it forces people to either duplicate the secrets in all the namespaces where they use ingresses or duplicate the issuers per namespace. In the end, the result is the same, it's a mess and you do end up propagating your certificate pretty much everywhere.

Contour and Ambassador both support namespaced TLS secret name to alleviate this issue.

**Comment by dtomcej** [comment]
@tlvenn, If a security flag was implemented in the provider to enable this functionality, and keep the current behavior as the default, I think both parties could be satisfied.

Namespace enforcement by default to keep with the kubernetes pattern
Cross-Namespace secret access if enabled via provider config to allow this.

**Comment by tlvenn** [comment]
From an security perspective there is an better approach taken by Contour I believe.

Their TLS delegation works in such a way that nobody can mount those secrets apart from Contour itself, so it's not like you are suddenly allowing any pod to mount those secrets from other namespaces:

![image](https://user-images.githubusercontent.com/3340262/68164287-30cf6e80-ffa8-11e9-8000-3ae28f28a095.png)

> https://github.com/projectcontour/contour/blob/master/design/tls-certificate-delegation.md

**Comment by JGiola** [comment]
There are any news on this issue? Is been considered to be implemented in the future or is a non goal for Traefik?

**Comment by deekue** [comment]
in case anyone else is stuck on this, the solution in #6057 in [this comment](https://github.com/traefik/traefik/issues/6057#issuecomment-615903064) worked for me.

TL;DR put the wildcard cert secret in kube-system and set it as the defaultCertificate in the default TLSStore

**Comment by chasebolt** [comment]
Also running into this issue when working with wild card certs for staging clusters. It is a mess trying to copy certs across many namespaces as new PRs are stood up by developers. Ideally, we would like to reference a single cert in a namespace dedicated for it.

**Comment by nmengin** [comment]
Hello there,

This proposal has been open for 6 months with minimal traction. We won't be adding it to our roadmap at this time and prefer to close it for now.

If circumstances change or interest grows, we can always reopen this later.
Thank you, @MikaelElkiaer  , for the suggestion!
